### DISCUSSION TITLE
Second I2C for WT32-SC01plus

### DISCUSSION THREAD

**Discussion by ARTMEDIUM** [lovyan03/LovyanGFX#573] [Q&A] [answered]
Hello.
I need advice from a more experienced users.
I'm making an Arduino project on WT32-SC01plus where the touch use I2C. Because ESP32S3 can use two I2C buses I tried to open the second I2C bus as Wire1 in order to connect my I2C device, but I see a conflict between my I2C device and the touch. At the same time, both the touch and my device work fine when the bus I2C of only one of them is turned on.
How can I solve my problem and is it possible to solve it at all?
Thanks for any information.
Georgy.

**Comment by tobozo** [comment]
hi,

please provide more info:

- what is your I2C device
- copy/paste the serial output showing that conflict
- a simple sketch showing how you use Wire1 and more importantly the LGFX configuration block you're using

**Reply by ARTMEDIUM** [reply]
Dear tobozo,

1. I2C device now is PCF8574. This is an I2C expander.

2. Serial output:
_PCF8574_LIB_VERSION:	0.4.1
could not initialize...
=> not connected_

3.simple sketch:
```cpp
#include "PanelLan.h"
#include "screen.h"
#include "screen2.h"
#include "screen3.h"
#include "Wire.h"
#include "PCF8574.h"

PanelLan tft(BOARD_SC01_PLUS);
PCF8574 PCF(0x27, &Wire1); 

int SDA1 = 10;
int SCL1 = 11;
int Freq1 = 100000;

static constexpr int image_width  = 480;
static constexpr int image_height = 320;
extern const uint16_t epd_bitmap_screen[];
extern const uint16_t epd_bitmap_screen2[];
extern const uint16_t epd_bitmap_screen3[];

void setup(void) {
  Serial.begin(115200);
  Serial.println(__FILE__);
  Serial.print("PCF8574_LIB_VERSION:\t");
  Serial.println(PCF8574_LIB_VERSION);
  
  tft.init();
  tft.initDMA();
  tft.begin();
  tft.pushImage(0, 0, image_width, image_height, (lgfx::rgb565_t*)epd_bitmap_screen);
  tft.setTextDatum(textdatum_t::middle_center);
  tft.setFont(&fonts::FreeSerifBold18pt7b);
  
  Wire1.begin(SDA1, SCL1, Freq1); 
  if (!PCF.begin())
  {
    Serial.println("could not initialize...");
  }
  if (!PCF.isConnected())
  {
    Serial.println("=> not connected");
    while(1);
  }
  int x = PCF.read8();
  Serial.print("Read ");
  Serial.println(x, HEX);
  delay(1000);
}

void loop(void) {
  int16_t x = 0;
  int16_t y = 0;

  if (tft.getTouch(&x, &y) > 0) {
      if(sq(x - 100) + sq(y-160) <= sq(100)) {
      tft.pushImage(0, 0, image_width, image_height, (lgfx::rgb565_t*)epd_bitmap_screen2);
      }
      if(sq(x - 380) + sq(y-160) <= sq(100)) {
      tft.pushImage(0, 0, image_width, image_height, (lgfx::rgb565_t*)epd_bitmap_screen3);
      }
      if((x>=140)&&(x<=340)&&(y>=160)&&(y<=260)) {
      tft.pushImage(0, 0, image_width, image_height, (lgfx::rgb565_t*)epd_bitmap_screen);
      }
    }
  
  PCF.toggle(5); 
  delay(20);
}
```
4.  PanelLan:   https://github.com/smartpanle/PanelLan_esp32_arduino

**Reply by tobozo** [reply, marked as answer]
according to [the board config](https://github.com/smartpanle/PanelLan_esp32_arduino/blob/master/src/board/sc01_plus/sc01_plus_pin.h) pins 10-11 are external pins so there's no conflict with pins assignation.

according to your board's [LGFX config](https://github.com/smartpanle/PanelLan_esp32_arduino/blob/master/src/board/sc01_plus/sc01_plus.cpp#L121C5-L121C30) the I2C bus for touch interface is using `I2C_NUM_1`

what you may try is to set it to `I2C_NUM_0` instead

:eyes: 

```cpp
  tft.init();
  tft.initDMA();
  tft.begin();
```

`tft.init()` is enough, no need to call initDMA() or begin() after that :-)

**Reply by yesnoj** [reply]
sorry for bring up this discussion, i'm having a similar issue.
I'm using the board from MakerFabs (https://www.makerfabs.com/esp32-s3-parallel-tft-with-touch-ili9488.html) that has 2 I2C connector behind.
If i configure the LovyanGFX with :

      cfg.i2c_port = I2C_PORT_NUM;      //    static constexpr int I2C_PORT_NUM = I2C_NUM_1;
      cfg.i2c_addr = I2C_LIB_INT;  
      cfg.pin_sda  = I2C_PIN_SDA;   
      cfg.pin_scl  = I2C_PIN_SCL;   

i can switch the current I2C modifing I2C_NUM_1 to I2C_NUM_0 and viceversa.
There is a way to use both?
Thanks in advance!
F

**Comment by ARTMEDIUM** [comment]
**YES!!!!!!!!!!!!!!!!!!!!!!!!!!!!**
_PCF8574_LIB_VERSION:	0.4.1
Read F7_
